hasher-matcher-actioner/hmalib/common/aws_dataclass.py

```diff
@@ -150,9 +161,9 @@ def aws_to_py(in_type: t.Type[T], aws_field: t.Any) -> T:
     if in_type is bool:  # BOOL
         return aws_field  # type: ignore # mypy/issues/10003
     if in_type is t.Set[str]:  # SS
-        return aws_field  # type: ignore # mypy/issues/10003
+        return {value for value in aws_field if value != EMPTY_SET_SENTINEL_VALUE_STR}  # type: ignore # mypy/issues/10003
```

Comment:
(ignorable): This compare might actually be costly for huge lists. You could directly check against the empty set case up front.

```
if origin is set and aws_field == {EMPTY_SET_SENTINEL_VALUE_STR}:
  return set()
```

Reply:
Valid point. Sets are hashed too. So should be 0(1) check in most cases. https://stackoverflow.com/questions/12390298/time-complexity-of-checking-if-two-set-are-equal-in-python